gbpservice/nfp/configurator/agents/loadbalancer_v1.py

```diff
@@ -0,0 +1,655 @@
+        """Handle the agent_updated notification event."""
+        arg_dict = {'context': context,
+                    'payload': payload}
+        ev = self._sc.new_event(id='AGENT_UPDATED', data=arg_dict)
+        self._sc.post_event(ev)
+
+
+"""Implements event handlers and their helper methods.
+
+Object of this class is registered with the event class of core service
+controller. Based on the event key, handle_event method of this class is
+invoked by core service controller.
+
+"""
+
+
+class LBaaSEventHandler(nfp_poll.PollEventDesc):
+    instance_mapping = {}
+
+    def __init__(self, sc, drivers, rpcmgr, nqueue):
+        self._sc = sc
+        self.drivers = drivers
+        self._rpcmgr = rpcmgr
+        self.nqueue = nqueue
+        self.plugin_rpc = LBaasRpcSender(sc)
+
+    def _get_driver(self, service_type):
+        """Retrieves service driver object based on service type input.
+
+        Currently, service drivers are identified with service type. Support
+        for single driver per service type is provided. When multi-vendor
+        support is going to be provided, the driver should be selected based
+        on both service type and vendor name.
+
+        :param service_type: Service type - loadbalancer
+
+        Returns: Service driver instance
+
+        """
+        service_type = 'loadbalancer'
+        return self.drivers[service_type]
+
+    def handle_event(self, ev):
+        """Processes the generated events in worker context.
+
+        Processes the following events.
+        - create pool
+        - update pool
+        - delete pool
+        - create vip
+        - update vip
+        - delete vip
+        - create member
+        - update member
+        - delete member
+        - create pool health monitor
+        - update pool health monitor
+        - delete pool health monitor
+        - agent updated
+        Enqueues responses into notification queue.
+
+        Returns: None
+
+        """
+
+        # Process single request data blob
+        try:
+            msg = ("Worker process with ID: %s starting "
+                   "to handle task: %s of topic: %s. "
+                   % (os.getpid(), ev.id, lb_constants.LBAAS_AGENT_RPC_TOPIC))
+            LOG.debug(msg)
+
+            method = getattr(self, "_%s" % (ev.id.lower()))
+            method(ev)
+        except Exception as err:
+            LOG.error("Failed to perform the operation: %s. %s"
+                      % (ev.id, str(err).capitalize()))
+            import traceback
+            traceback.print_exc()
+        """
+        finally:
+            if ev.id == 'COLLECT_STATS':
+                '''Do not say event done for collect stats as it is
+                   to be executed forever
+                '''
+                pass
+            else:
+                self._sc.event_done(ev)
+        """
+
+    def _create_vip(self, ev):
+        data = ev.data
+        context = data['context']
+        vip = data['vip']
```

Comment:
Most of the code in these methods is repetitive. Can we have one function for each resource? Like handle_event_vip which covers CRUD of VIP and same with other resources?